synapse/rest/admin/federation.py

```diff
@@ -111,12 +111,26 @@ async def on_GET(
     ) -> Tuple[int, JsonDict]:
         await assert_requester_is_admin(self._auth, request)
 
+        if not await self._store.is_destination_known(destination):
+            raise NotFoundError("Unknown destination")
+
         destination_retry_timings = await self._store.get_destination_retry_timings(
             destination
         )
 
-        if not destination_retry_timings:
-            raise NotFoundError("Unknown destination")
+        retry_timing_response: JsonDict = {}
```

Comment:
It might be clearer to create the dictionary initially with `destination` and then add more keys to it, but I don't feel strongly either way!

Reply:
Your suggestion is response (`GET /_synapse/admin/v1/federation/destinations/<destination>`) with
```json
{
  "destination": {
     "destination": "matrix.org",
     "retry_last_ts": 1557332397936,
     "retry_interval": 3000000,
     "failure_ts": 1557329397936,
     "last_successful_stream_ordering": null
  }
}
```

instead of:
```json
{
   "destination": "matrix.org",
   "retry_last_ts": 1557332397936,
   "retry_interval": 3000000,
   "failure_ts": 1557329397936,
   "last_successful_stream_ordering": null
}
```
?

This would be different to rooms and users.
- https://matrix-org.github.io/synapse/latest/admin_api/user_admin_api.html
- https://matrix-org.github.io/synapse/latest/admin_api/rooms.html#room-details-api

Reply:
No, my suggestion is to build the response as:

```python
retry_timing_response: JsonDict = {
    "destination": destination,
}
retry_timing_response[...] = ...
```